hw/ip/otbn/doc/_index.md

```diff
@@ -361,6 +361,66 @@ Both use the same state for tracking control flow.
 This is a stack of tuples containing a loop count, start address and end address.
 The stack has a maximum depth of eight and the top of the stack is the current loop.
 
+### Loop nesting
+
+OTBN permits nested loops and branch and jumps inside loops.
+The programmer must ensure loops are *well nested*.
+For a loop to be *well nested* the following conditions must be met:
+
+* Control flow of a loop must eventually return to its end instruction without passing the end instruction of any outer loop.
+* Nested loops cannot end on the same instruction.
+
+The following loops are *well nested*:
```

Comment:
Maybe it would be a good idea to say something explicit about the fact that the indentation in the code examples is just for clarity, and has no functional impact.